doc/design/publish-1.md

```diff
@@ -0,0 +1,141 @@
+# Moving Parts
+
+This is a summary of system components that are relevant to publish and how they relate to eachother.
+
+### Dandiset
+A DANDI dataset.
+Each dandiset is guaranteed to have a draft Version, and can also have any number of published Versions.
+
+### Version
+Versions consist of a JSON blob of metadata and any number of Assets.
+There are two kinds of Version: draft and published.
+Published Versions are immutable, only draft Versions are editable (except admins, who have superpowers).
+Versions also track their validation status.
```

Comment:
`### Version` is not under `### Dandiset` so a bit confusing here on which metadata (above suggestion) or validation we are talking about:  validation of dandiset record (metadata), of each asset, or a composite ( everything pertinent to Version: dandiset metadata + all assets metadata, etc).  If I got it right may be:
```suggestion
Versions also track their validation status (comprised of validation of dandiset and assets metadata validation, etc).
```

Reply:
As currently implemented, Versions only track the validation of their own metadata, not their Assets. You could have a `VALID` version with `INVALID` assets. When a publish is attempted, the `INVALID` assets are checked, so for publishing purposes the version is considered `INVALID`, despite having a column that is marked `VALID`.

```suggestion
Versions also track the validation status of their Dandiset metadata. Asset metadata validation is not recorded in the Version.
```